test/built-ins/RegExp/named-groups/groups-object.js

```diff
@@ -0,0 +1,34 @@
+// Copyright 2017 Aleksey Shvayka. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+description: Properties of the groups object are created with CreateDataProperty
+includes: [propertyHelper.js]
+esid: sec-regexpbuiltinexec
+features: [regexp-named-groups]
+info: >
+  Runtime Semantics: RegExpBuiltinExec ( R, S )
+    24. If R contains any GroupName,
+      a. Let groups be ObjectCreate(null).
+      b. Perform ! CreateDataProperty(A, "groups", groups).
+---*/
+
+// groups is created with Define, not Set
+let counter = 0;
+Object.defineProperty(Array.prototype, "groups", {set() { counter++; }});
+let match = /(?<x>.)/.exec("a");
+assert.sameValue(counter, 0);
+
+// groups is writable, enumerable and configurable
+// (from CreateDataProperty)
+verifyProperty(match, "groups", {
+  writable: true,
+  enumerable: true,
+  configurable: true,
+});
+
+// The '__proto__' property on the groups object is not special,
+// and does not affect the [[Prototype]] of the resulting groups object.
+let {groups} = /(?<__proto__>.)/.exec("a");
+assert.sameValue("a", groups.__proto__);
```

Comment:
I believe this test belongs here, but it feels weird the counter effects lives in the Annex B folder as `__proto__` is not specified out of it.

Reminds me TC39 should continue discussing if Annex B should be merged.